modules/core/04-channel/keeper/export_test.go

```diff
@@ -0,0 +1,23 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+)
+
+// StartFlushUpgradeHandshake is an exported version of startFlushUpgradeHandshake to be used for unit testing
+func (k Keeper) StartFlushUpgradeHandshake(
```

Comment:
ongoing discussion on this construction [here](https://github.com/cosmos/ibc-go/pull/3687#discussion_r1209408650), but needed to run unit tests and keep this function private